app/src/main/java/com/codesoom/assignment/exceptions/NotFoundTaskException.java

```diff
@@ -0,0 +1,16 @@
+package com.codesoom.assignment.exceptions;
+
+/**
+ * 주어진 Id에 해당하는 할 일이 목록에 없을 경우 발생하는 예외입니다.

```

Comment:
이 설명을 잘 읽어봅시다. 그리고 클래스의 이름을 잘 읽어봅시다. 클래스의 이름을 읽어보면 이 예외의 설명은 다음과 같이 수정할 수 있어요.

```suggestion
 * 할 일을 찾을 수 없는 경우에 던집니다.
```

예외는 던지는 거죠. "주어진 Id에 해당하는..."에 대한 설명은 아래에 있는 생성자에 달아주는 편이 더 적절해 보입니다.

Reply:
1, 대체로 맞지만 약간 다릅니다. 저는 @DevRunner21 님이 작성한 예외 클래스의 이름이 특정한 그룹의 예외를 적절히 추상화해냈다고 생각했어요. `NotFoundTaskException`은 할 일 목록과 관계 없이 할 일을 찾을 수 없는 경우에 던질 의도로 만들어진 것으로 보이거든요. 클래스 이름을 보면 `할 일을 할 일 목록에서 찾았지만 찾지 못했`거나, `할 일을 할 일 DB에서 찾아봤지만 찾지 못했`거나, `할 일을 할 일 Tape Archive에서 찾아봤지만 찾지 못했`거나, `할 일을 할 일을 의미하는 노트 필기를 스캔한 이미지 파일에서 찾지 못했`거나 등 굉장히 다양한 경우의 예외 상황에 모두 사용할 수 있는 추상화된 예외입니다.

@DevRunner21 님은 그러한 추상화를 한 예외를 정의한 거에요. 이 프로그램이 앞으로 얼마나 더 발전하고, 얼마나 많은 사용자가 다양한 목적으로 사용하게 될 지는 아직은 알 수 없습니다. 만약 이름이 `TaskNotFoundFromTaskListException`이라면 할 일 목록에서 찾을 때에만 사용할 수 있는 너무 구체적인 예외가 되었겠죠.

2, 주석을 삭제하는 것이 아쉽다면 이 설명을 그대로 옮겨 생성자에 달아줄 수도 있을 거라는 의미에요. 생성자의 시그니처를 보면 다음과 같습니다.

```java
public NotFoundTaskException(Long taskId) {
```

여기에 이렇게 작성해줄 수 있다는 거죠. 이 주석은 `id`를 명시하고 있고, 생성자의 시그니처는 `taskId`를 알고 있습니다.

```java
/** 주어진 Id에 해당하는 할 일이 목록에 없을 경우 발생하는 예외를 생성합니다. */
public NotFoundTaskException(Long taskId) {
```